web/lib/tasks/pictograms.rake

```diff
@@ -1,11 +1,96 @@
+require "open-uri"
+
 namespace :pictograms do
+  desc 'Populate pictograms table with images from a folder'
+  task :populate_table, [:folder] => :environment do |task, args|
+    folder = args.folder
+    unless folder
+      msg = 'You need specify a folder param (e.g. rails pictograms:populate_table[<folder>]).'
+      abort(msg)
+    end
+    populate_table(folder)
+  end
+
   desc 'Download some pictograms from araasac and populate db'
   task download_samples: :environment do
-    require "open-uri"
+    download_samples()
+  end
+
+
+  # functions
+
+  def populate_table(folder)
+    entries = Dir.entries(folder).reject { |entry| File.directory? entry }
+    validated_filetypes = check_file_type(folder, entries)
+    normalized_filenames = normalize_filenames(validated_filetypes)
+    unique_filenames = generate_unique_filenames(normalized_filenames)
+    rename_files(folder, entries, unique_filenames)
+    insert_into_pictogram_table(folder)
+
+  end
+
+  def check_file_type(folder, entries)
```

Comment:
You're not just 'checking' type file, right? You'are changing the extension if needed.
Maybe changing the methods name to 'handle_file_type' or something?